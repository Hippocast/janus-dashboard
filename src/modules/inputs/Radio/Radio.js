import React from 'react';

import block from '../../../helpers/bem-cn';

import './Radio.css';

const radio = block('j-radio');

const Radio = () => {
    return (
        <div className={radio('wrap')}>
            <input type="radio" name="testRadio" id="testRadio-2" className={radio('input')} />
            <div className={radio('d1')}>
                <div className={radio('d2')}>
                    <div>
                        <svg className={radio('svg', { '1': true })} viewBox="0 0 24 24">
                            <path className={radio('path')} d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"></path>
                        </svg>
                        <svg className={radio('svg', { '2': true })}viewBox="0 0 24 24" >
                            <path className={radio('path')} d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"></path>
                        </svg>
                    </div>
                </div>
                <label className="j-m-radio__label" htmlFor="testRadio-2">Label 1</label>
            </div>
        </div>
    );
};

export default Radio;
