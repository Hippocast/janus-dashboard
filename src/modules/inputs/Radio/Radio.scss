@import 'index.scss';

.j-radio {
    &__wrap {
        position: relative;
        cursor: pointer;
    }

    &__input {
        position: absolute;
        cursor: pointer;
        pointer-events: all;
        opacity: 0;
        width: 100%;
        height: 100%;
        z-index: 2;
        left: 0px;
        box-sizing: border-box;
        padding: 0px;
        margin: 0px;

        &:checked + .j-radio__d1 {
            .j-radio__svg--1 {
                opacity: 0;
                transform: scale(0);
            }

            .j-radio__svg--2 {
                transform: scale(1);
            }
        }
    }

    &__d2 {
        transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
        position: relative;
        display: block;
        flex-shrink: 0;
        width: 24px;
        height: 24px;
    }

    &__svg {
        &--1 {
            display: inline-block;
            color: rgba(0, 0, 0, 0.870588);
            fill: rgba(0, 0, 0, 0.870588);
            height: 24px;
            width: 24px;
            user-select: none;
            transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
            position: absolute;
            opacity: 1;
            transform: scale(1);
        }

        &--2 {
            display: inline-block;
            color: rgba(0, 0, 0, 0.870588);
            fill: rgb(0, 188, 212);
            height: 24px;
            width: 24px;
            user-select: none;
            transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
            position: absolute;
            opacity: 1;
            transform: scale(0);
            transform-origin: 50% 50% 0px;
        }
    }

    &__path {
        fill: $hellofresh-color;
    }
}
