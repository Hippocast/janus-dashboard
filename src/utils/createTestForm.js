import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';

const Form = ({ children, initialValues }) => children;

const createTestForm = initialValues => connect(
    () => ({
        initialValues
    }),
    null,
)(
    reduxForm({
        form: 'mockForm',
    })(Form)
);

export default createTestForm;
