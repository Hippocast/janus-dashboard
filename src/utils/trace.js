import R from 'ramda';
import { openResponseModal } from '../store/actions';

/**
 * @name trace
 * @desc helper function for development only.
 *          It should be used like this:
 * @example (from https://github.com/MostlyAdequate/mostly-adequate-guide/blob/master/ch5.md#debugging)
 *
 * import trace from '../utilz/trace';
 *
 * const dasherize = R.compose(
 *      join('-'),
 *      toLower,
 *      trace('after split'),
 *      split(' '),
 *      replace(/\s{2,}/ig, ' ')
 * );
 *
 * dasherize('The world is a vampire');
 * // after split [ 'The', 'world', 'is', 'a', 'vampire' ]
 */
const trace = R.curry((tag, x) => {
    console.error('trace -> ', tag, x);
    return x;
});

export default trace;
