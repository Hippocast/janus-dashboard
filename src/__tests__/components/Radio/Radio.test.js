import React from 'react';
import { createStore } from 'redux';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { renderFakeForm, wrap } from '../../../utils/createTestForm';
import Radio from '../../../components/Radio/Radio';

const initialValues = {
    'mock-name': '',
};

const store = createStore(() => ({
    form: {
        mockForm: {
            values: initialValues,
        }
    }
}));

describe('Radio component', () => {
    const requiredProps = {
        name: 'mock-name',
        id: 'mock-id',
        input: {
            checked: true,
        }
    };

    it('renders correctly', () => {
        const tree = wrap(store)(initialValues)(
            <Radio
                {...requiredProps}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
