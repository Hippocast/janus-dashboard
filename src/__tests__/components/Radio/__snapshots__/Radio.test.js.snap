// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component renders correctly 1`] = `
<div
  className="j-radio__wrap"
>
  <input
    className="j-radio__input"
    id={undefined}
    name="mock-name"
    type="radio"
  />
  <div
    className="j-radio__d1"
  >
    <div
      className="j-radio__d2"
    >
      <div>
        <svg
          className="j-radio__svg j-radio__svg--1"
          viewBox="0 0 24 24"
        >
          <path
            className={[Function]}
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          className="j-radio__svg j-radio__svg--2"
          viewBox="0 0 24 24"
        >
          <path
            className={[Function]}
            d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
